Generate an App V3 resource reference page (#62163)
gravitational/teleport: docs/pages/reference/infrastructure-as-code/teleport-resources/app-v3.mdx

sub_kind: "string"
version: "string"
metadata: # [...]
spec: # [...]
```
|Field Name|Description|Type|
|---|---|---|
|kind|The app resource kind. Always "app".|string|
|metadata|The app resource metadata.|[Metadata](#metadata)|
|spec|The app resource spec.|[App Spec V3](#app-spec-v3)|
|sub_kind|An optional resource subkind.|string|
|version|The resource version. It must be specified. Supported values are:`v3`.|string|

## App AWS

Contains additional options for AWS applications.

Example:

```yaml
external_id: "string"
roles_anywhere_profile: # [...]
```

|Field Name|Description|Type|
|---|---|---|
|external_id|The AWS External ID used when assuming roles in this app.|string|
|roles_anywhere_profile|Contains the IAM Roles Anywhere fields associated with this Application. These fields are set when performing the synchronization of AWS IAM Roles Anywhere Profiles into Teleport Apps.|[App AWS Roles Anywhere Profile](#app-aws-roles-anywhere-profile)|

## App AWS Roles Anywhere Profile

Contains the fields that represent an AWS IAM Roles Anywhere Profile.

Example:

```yaml
profile_arn: "string"
accept_role_session_name: true
```

|Field Name|Description|Type|
|---|---|---|

> |accept_role_session_name|Whether this Roles Anywhere Profile accepts a custom role session name. When not supported, the AWS Session Name will be the X.509 certificate's serial number. When supported, the AWS Session Name will be the identity's username. This values comes from: https://docs.aws.amazon.com/rolesanywhere/latest/APIReference/API_ProfileDetail.html / acceptRoleSessionName|Boolean|

Reviewer: ```suggestion
|accept_role_session_name|Whether this Roles Anywhere Profile accepts a custom role session name. When not supported, the AWS Session Name will be the X.509 certificate's serial number. When supported, the AWS Session Name will be the identity's username. This value comes from: https://docs.aws.amazon.com/rolesanywhere/latest/APIReference/API_ProfileDetail.html / acceptRoleSessionName|Boolean|
```

|account_id|Account ID is the AWS-assigned ID of the account|string|
|permission_sets|Lists the available permission sets on the given account|[][Identity Center Permission Set](#identity-center-permission-set)|

## App Spec V3

The AppV3 resource spec.

Example:

```yaml
uri: "string"
public_addr: "string"
dynamic_labels: 
  "string": # [...]
  "string": # [...]
  "string": # [...]
insecure_skip_verify: true
rewrite: # [...]
aws: # [...]
cloud: "string"
UserGroups: 
  - "string"
  - "string"
  - "string"
integration: "string"
required_app_names: 
  - "string"
  - "string"
  - "string"
cors: # [...]
identity_center: # [...]
tcp_ports: 
  - # [...]
  - # [...]
  - # [...]
use_any_proxy_public_addr: true
mcp: # [...]
```

|Field Name|Description|Type|
|---|---|---|
|UserGroups|A list of user group IDs that this app is associated with.|[]string|
|aws|Contains additional options for AWS applications.|[App AWS](#app-aws)|
|cloud|Identifies the cloud instance the app represents.|string|
|cors|Policy defines the Cross-Origin Resource Sharing settings for the app.|[CORS Policy](#cors-policy)|
|dynamic_labels|The app's command labels.|map[string][Command Label V2](#command-label-v2)|

> |identity_center|Encasulates AWS identity-center specific information. Only valid for Identity Center account apps.|[App Identity Center](#app-identity-center)|

Reviewer: ```suggestion
|identity_center|Encapsulate AWS identity-center specific information. Only valid for Identity Center account apps.|[App Identity Center](#app-identity-center)|
```

|insecure_skip_verify|Disables app's TLS certificate verification.|Boolean|
|integration|The integration name that must be used to access this Application. Only applicable to AWS App Access. If present, the Application must use the Integration's credentials instead of ambient credentials to access Cloud APIs.|string|
|mcp|Contains MCP server related configurations.|[MCP](#mcp)|
|public_addr|The public address the application is accessible at.|string|

> |required_app_names|A list of app names that are required for this app to function. Any app listed here will be part of the authentication redirect flow and authenticate along side this app.|[]string|

Reviewer: ```suggestion
|required_app_names|A list of app names that are required for this app to function. Any app listed here will be part of the authentication redirect flow and authenticate alongside this app.|[]string|
```

Example:

```yaml
arn: "string"
name: "string"
assignment_name: "string"
```

|Field Name|Description|Type|
|---|---|---|
|arn|The fully-formed ARN of the Permission Set.|string|
|assignment_name|The ID of the Teleport Account Assignment resource that represents this permission being assigned on the enclosing Account.|string|
|name|The human-readable name of the Permission Set.|string|

## MCP

Contains MCP server-related configurations.

Example:

```yaml
command: "string"
args: 
  - "string"
  - "string"
  - "string"
run_as_host_user: "string"
```

|Field Name|Description|Type|
|---|---|---|
|args|To execute with the command.|[]string|
|command|To launch stdio-based MCP servers.|string|
|run_as_host_user|The host user account under which the command will be executed. Required for stdio-based MCP servers.|string|

## Metadata

Resource metadata

Example:

```yaml
name: "string"
description: "string"
labels: 
  "string": "string"
  "string": "string"
  "string": "string"
expires: # See description
revision: "string"
```

|Field Name|Description|Type|
|---|---|---|
|description|Object description|string|

> |expires|A global expiry time header can be set on any resource in the system.||

Reviewer: The Type field is empty for `expires` - should the value be `time (RFC 3339)`?
